src/app/state/playback_state.rs

```diff
@@ -407,13 +424,21 @@ impl From<PlaybackAction> for AppAction {
 pub enum PlaybackEvent {
     PlaybackPaused,
     PlaybackResumed,
+    RepeatModeChanged(RepeatMode),
     TrackSeeked(u32),
     SeekSynced(u32),
     TrackChanged(String),
     PlaylistChanged,
     PlaybackStopped,
 }
 
+#[derive(Clone, Debug)]
```

Comment:
just thought of this but you can make it Copy as well to avoid explicit .clone()